Bottom: c3252da09e1e769e714f5e9969503cce1df221a6
Top:    09ed451c6b5c30d2dbe0bbedc3379ef8034dc899
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2014-04-09 22:29:34 -0700

Refresh of lazy-init

---

diff --git a/src/cma.c b/src/cma.c
index 892d69e..0dc229e 100644
--- a/src/cma.c
+++ b/src/cma.c
@@ -336,7 +336,7 @@ int ucma_init_all(void)
 
 	pthread_mutex_lock(&mut);
 	for (i = 0; i < cma_dev_cnt; i++) {
-		ret = ucma_init_device(ucma_dev_array[i]);
+		ret = ucma_init_device(&cma_dev_array[i]);
 		if (ret)
 			break;
 	}
@@ -346,7 +346,7 @@ int ucma_init_all(void)
 
 struct ibv_context **rdma_get_devices(int *num_devices)
 {
-	struct ibv_context **devs;
+	struct ibv_context **devs = NULL;
 	int i;
 
 	if (ucma_init_all())
@@ -416,7 +416,7 @@ static int ucma_get_device(struct cma_id_private *id_priv, uint64_t guid)
 
 	return ERR(ENODEV);
 match:
-	if ((ret = ucma_dev_init(cma_dev)))
+	if ((ret = ucma_init_device(cma_dev)))
 		return ret;
 
 	pthread_mutex_lock(&mut);
